2023-04-30 22:51:06,689 - __init__.py:init_device[line:18] - CRITICAL: 初始化网关设备驱动 start...
2023-04-30 22:51:06,702 - device_manager.py:open[line:64] - INFO: SUCCESS : open device设备('水声通信机0')
2023-04-30 22:51:06,702 - core_scheduling.py:init_device_list_threadhandle[line:51] - INFO: 设备列表中存在1个设备, 设备0=设备('水声通信机0')
2023-04-30 22:51:06,709 - connection_workflow.py:start[line:179] - INFO: Pika version 1.3.1 connecting to ('::1', 5672, 0, 0)
2023-04-30 22:51:06,710 - io_services_utils.py:_on_writable[line:345] - INFO: Socket connected: <socket.socket fd=96, family=AddressFamily.AF_INET6, type=SocketKind.SOCK_STREAM, proto=6, laddr=('::1', 7569, 0, 0), raddr=('::1', 5672, 0, 0)>
2023-04-30 22:51:06,756 - connection_workflow.py:_on_transport_establishment_done[line:428] - INFO: Streaming transport linked up: (<pika.adapters.utils.io_services_utils._AsyncPlaintextTransport object at 0x0000027C9BEBF490>, _StreamingProtocolShim: <SelectConnection PROTOCOL transport=<pika.adapters.utils.io_services_utils._AsyncPlaintextTransport object at 0x0000027C9BEBF490> params=<ConnectionParameters host=localhost port=5672 virtual_host=/ ssl=False>>).
2023-04-30 22:51:06,759 - connection_workflow.py:_report_completion_and_cleanup[line:293] - INFO: AMQPConnector - reporting success: <SelectConnection OPEN transport=<pika.adapters.utils.io_services_utils._AsyncPlaintextTransport object at 0x0000027C9BEBF490> params=<ConnectionParameters host=localhost port=5672 virtual_host=/ ssl=False>>
2023-04-30 22:51:06,759 - connection_workflow.py:_report_completion_and_cleanup[line:725] - INFO: AMQPConnectionWorkflow - reporting success: <SelectConnection OPEN transport=<pika.adapters.utils.io_services_utils._AsyncPlaintextTransport object at 0x0000027C9BEBF490> params=<ConnectionParameters host=localhost port=5672 virtual_host=/ ssl=False>>
2023-04-30 22:51:06,759 - blocking_connection.py:_create_connection[line:453] - INFO: Connection workflow succeeded: <SelectConnection OPEN transport=<pika.adapters.utils.io_services_utils._AsyncPlaintextTransport object at 0x0000027C9BEBF490> params=<ConnectionParameters host=localhost port=5672 virtual_host=/ ssl=False>>
2023-04-30 22:51:06,760 - blocking_connection.py:__init__[line:1264] - INFO: Created channel=1
2023-04-30 22:51:06,761 - core_scheduling.py:monitor_web_task_threadhandle[line:216] - CRITICAL: SUCCESS : Create Pika server.
2023-04-30 22:51:21,550 - bsp_underwater_acoustic_comm.py:_write_bytes[line:185] - CRITICAL: 水声通信设备(端口'COM5' 状态True)传输一帧数据b'U\xaa6\x01test.txt\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\r\n'
2023-04-30 22:51:23,244 - bsp_underwater_acoustic_comm.py:_write_bytes[line:190] - WARNING: 水声通信设备(端口'COM5' 状态True) 发送数据时接收到0X66 0XBB确认回馈
2023-04-30 22:51:24,989 - bsp_underwater_acoustic_comm.py:_read_bytes[line:209] - CRITICAL: 水声通信设备(端口'COM5' 状态True) 接收到一帧数据b'U\xaa6\x04filename_received\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\r\n'
2023-04-30 22:51:24,993 - bsp_underwater_acoustic_comm.py:_write_bytes[line:185] - CRITICAL: 水声通信设备(端口'COM5' 状态True)传输一帧数据b'U\xaa6\x0238\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\r\n'
2023-04-30 22:51:26,193 - bsp_underwater_acoustic_comm.py:_write_bytes[line:190] - WARNING: 水声通信设备(端口'COM5' 状态True) 发送数据时接收到0X66 0XBB确认回馈
2023-04-30 22:51:27,121 - bsp_underwater_acoustic_comm.py:_read_bytes[line:209] - CRITICAL: 水声通信设备(端口'COM5' 状态True) 接收到一帧数据b'U\xaa6\x04filesize_received\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\r\n'
2023-04-30 22:51:27,125 - bsp_underwater_acoustic_comm.py:_write_bytes[line:185] - CRITICAL: 水声通信设备(端口'COM5' 状态True)传输一帧数据b'U\xaa6\x0301234567890123456789012345678901234567\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\r\n'
2023-04-30 22:51:28,892 - bsp_underwater_acoustic_comm.py:_write_bytes[line:190] - WARNING: 水声通信设备(端口'COM5' 状态True) 发送数据时接收到0X66 0XBB确认回馈
2023-04-30 22:51:55,133 - bsp_underwater_acoustic_comm.py:_read_bytes[line:209] - CRITICAL: 水声通信设备(端口'COM5' 状态True) 接收到一帧数据b'U\xaa6\x01fiieaaaaQreceived\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\r\n'
2023-04-30 22:51:55,138 - bsp_underwater_acoustic_comm.py:_write_bytes[line:185] - CRITICAL: 水声通信设备(端口'COM5' 状态True)传输一帧数据b'U\xaa6\x04filename_received\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\r\n'
2023-04-30 22:51:56,689 - bsp_underwater_acoustic_comm.py:_write_bytes[line:190] - WARNING: 水声通信设备(端口'COM5' 状态True) 发送数据时接收到0X66 0XBB确认回馈
2023-04-30 22:51:57,841 - bsp_underwater_acoustic_comm.py:_read_bytes[line:209] - CRITICAL: 水声通信设备(端口'COM5' 状态True) 接收到一帧数据b'U\xaa6\x02123\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\r\n'
2023-04-30 22:51:57,845 - bsp_underwater_acoustic_comm.py:_write_bytes[line:185] - CRITICAL: 水声通信设备(端口'COM5' 状态True)传输一帧数据b'U\xaa6\x04filesize_received\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\r\n'
2023-04-30 22:51:58,829 - bsp_underwater_acoustic_comm.py:_write_bytes[line:190] - WARNING: 水声通信设备(端口'COM5' 状态True) 发送数据时接收到0X66 0XBB确认回馈
2023-04-30 22:51:59,877 - bsp_underwater_acoustic_comm.py:_read_bytes[line:209] - CRITICAL: 水声通信设备(端口'COM5' 状态True) 接收到一帧数据b'U\xaa6\x03123\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\r\n'
2023-04-30 22:52:02,044 - bsp_underwater_acoustic_comm.py:_read_bytes[line:209] - CRITICAL: 水声通信设备(端口'COM5' 状态True) 接收到一帧数据b'U\xaa6\x03123\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\r\n'
2023-04-30 22:52:03,336 - bsp_underwater_acoustic_comm.py:_read_bytes[line:209] - CRITICAL: 水声通信设备(端口'COM5' 状态True) 接收到一帧数据b'U\xaa6\x03123\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\r\n'
2023-04-30 22:52:03,338 - _base.py:_invoke_callbacks[line:331] - ERROR: exception calling callback for <Future at 0x27c9bed5a90 state=finished raised AttributeError>
Traceback (most recent call last):
  File "Y:\研究生\跨越网关\代码\Gateway_System\control\src\core_scheduling.py", line 95, in deal_data_forwarding_processhandle
    transfer_speed  = str(round(float(file_size) / float(cal_diff_time_between_date1_and_date2(execute_time, finish_time)),2))
UnboundLocalError: local variable 'file_size' referenced before assignment

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "S:\anaconda3\lib\concurrent\futures\_base.py", line 329, in _invoke_callbacks
    callback(self)
  File "Y:\研究生\跨越网关\代码\Gateway_System\control\src\core_scheduling.py", line 132, in deal_data_forwarding_callback
    task = res.result()
  File "S:\anaconda3\lib\concurrent\futures\_base.py", line 438, in result
    return self.__get_result()
  File "S:\anaconda3\lib\concurrent\futures\_base.py", line 390, in __get_result
    raise self._exception
  File "S:\anaconda3\lib\concurrent\futures\thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "Y:\研究生\跨越网关\代码\Gateway_System\control\src\core_scheduling.py", line 106, in deal_data_forwarding_processhandle
    logging.error(f"ERROR   : 线程池处理DIRECT_DEVICE_TO_FILE任务{task.task_id}")
AttributeError: 'mTask' object has no attribute 'task_id'
